Finished Game with score and recolectables

Rokziell opened the pull request:
Finished Game

Rokziell committed:
Version 1

Finished Game

charly-palencia commented:
@Rokirito  pleases change your commit name. this is not the version 1, you should use a more descriptive name for your commit.

example `Implement Coins objects and GameOver functionality`

charly-palencia commented on src/scenes/boot-scene.js:
```diff
@@ -1,40 +1,62 @@
+    this.diamonds = this.addAllDiamonds();
+    this.gameOverSound = this.sound.add("gameOverSound");
+    this.keyR = this.input.keyboard.addKey("R");
+    this.score = 0;
```
all this logic about the score should be move it into a method

Rokziell replied:
all the logic about the score is inside a method called addScore()

charly-palencia commented on src/scenes/boot-scene.js:
```diff
@@ -1,40 +1,62 @@
+    }
+
+    if(this.redBall.isDead){
+      this.rToReset();
```
what is a rToReset? please use a better name for this `resetGame` could be an option

Rokziell replied:
make the name's change to resetGame()

charly-palencia commented on src/scenes/boot-scene.js:
```diff
@@ -45,5 +67,64 @@ export default class BootScene extends Phaser.Scene {
+    diamond.destroy();
+  }
+
+  addGameOver(){
```
It's a good idea to split our method but you need to have for that. in this case `gameOver` is a functionality itself so you can create some method like `runGameOver` and then include the addGameOver, addGameOverSound and addFameOverMessage into it (also the `this.isGameOver = true;` in line 52). we don't need three different methods to understand what runGameOver method is doing

Rokziell replied:
Ok, I created a method called runGameOver that include addGameOver, addGameOverSound,  addGameOverMessage and the "this.isGameOver = true;" and put the method runGameOver inside the if(this.redBall is dead && !this.isGameOver)

charly-palencia commented:
@Rokirito  Include more feature details in this PR. for example what changes are you doing?

Rokziell committed:
Finished Video Game

Included:
-Colectables
-Score
-Game Over Screen
-Added the "R" key to reset the game
